src/lib/Workflows.ts

```diff
@@ -248,17 +247,25 @@ export class Workflows {
       }
     }
 
+    // Save or update workflows from the server
     for (const wf of wfsFromSrv) {
-      const fileName = getFileName(wf);
-      const filePath = path.join(dir, fileName);
-
+      const newFileName = getFileName(wf);
+      const newFilePath = path.join(dir, newFileName);
+  
+      // Check if there is already a workflow with the same ID but a different name
+      const existingFile = filesList.find(file => getIdFromFileName(file) === wf.id);
+  
+      if (existingFile && existingFile !== newFileName) {
+        // If workflow with the same ID but a different name exists, remove the old file
+        fs.unlinkSync(path.join(dir, existingFile));
+      }
```

Comment:
Shouldn't this `if` statement with action be inside following the `if`? Otherwise it will be possible when we delete old file, but not create new one.

Reply:
You're right. I'll move the if statement inside to ensure a new file is created after deleting the old one. Thanks!